src/gui/vector/qgsdiagramproperties.cpp

```diff
@@ -1118,3 +1245,71 @@ void QgsDiagramProperties::createAuxiliaryField()
 
   emit auxiliaryFieldCreated();
 }
+
+void QgsDiagramProperties::connectValueChanged( const QList<QWidget *> &widgets, const char *signal )
+{
+  const auto constWidgets = widgets;
+  for ( QWidget *widget : constWidgets )
+  {
+    if ( QgsSymbolButton *w = qobject_cast<QgsSymbolButton *>( widget ) )
+    {
+      connect( w, SIGNAL( changed() ), this, signal );
```

Comment:
Can you switch theses to functor based connections?

Reply:
Yes, should I also adjust https://github.com/qgis/QGIS/blob/da1f55b7d1ca1026535ebadb39303df4447cc729/src/gui/symbology/qgsrendererpropertiesdialog.cpp#L140 and https://github.com/qgis/QGIS/blob/da1f55b7d1ca1026535ebadb39303df4447cc729/src/gui/qgstextformatwidget.cpp#L662 for consistency?

Reply:
Done in https://github.com/qgis/QGIS/pull/58568/commits/01055d7a2d48f999ea193410fec7e5764d5135cc and https://github.com/qgis/QGIS/pull/58568/commits/eb201c56e61919844b21659f9c3493d3a6441332